packages/frontend/app/models/vote-chain.ts

```diff
@@ -0,0 +1,18 @@
+import Model, { attr, hasMany } from '@ember-data/model';
+
+import Identity from 'emberclear/models/identity';
+
+export default class VoteChain extends Model {
+    @hasMany('identity', { async: false }) remaining!: Identity[];
+    @hasMany('identity', { async: false }) yes!: Identity[];
+    @hasMany('identity', { async: false}) no!: Identity[];
+    @attr() previousVoteKey!: Identity;
+    @attr() previousVoteChain!: VoteChain;//declare as encrypted
```

Comment:
When ember tries to convert data to a model, is there a way to mark something as encrypted? If not, can you declare the type as just raw data so we can could have a encrypted field with just data. @NullVoxPopuli basically when we send a vote it will be encrypted and each layer within it will also be encrypted.

Reply:
the type is less for semantics, and more about describing the data. 

for example: over here, where Uint8Arrays are encrypted, it's still a Uint8Array: https://github.com/NullVoxPopuli/emberclear/blob/master/packages/frontend/app/workers/crypto/utils/nacl.ts#L43

if the field ever needs to be anything _other_ than _always_ encrypted, you may want to look at managing the value in a state machine.

properties should be single-purpose, and not sometimes be encrypted or sometimes not :)

> When ember tries to convert data to a model, is there a way to mark something as encrypted? 

fyi, this is the method in charge of transforming data:
https://github.com/NullVoxPopuli/emberclear/blob/master/packages/frontend/app/serializers/application.js#L59

Reply:
Would you recommend storing a fully signed one and one where the layers have been decrypted and its just the data we need? We definitely need the signed one so that when a user wants to vote they can take the previously signed vote, add their own vote, and sign it.

Reply:
> storing a fully signed one

fully signed what?

> where the layers have been decrypted and its just the data we need?

data should be fully decrypted once it it deserialized from JSON in the messaging/network layer -- I don't think it makes sense to keep things encrypted in the data/model layer - even for signature verification -- just do that when receiving the message -- though for the chain of signature's that's different -- you could keep the signatures, but you wouldn't need to re-(de|en)crypt unless you need to rebuild the chain for some reason, I think.
idk, I could be wrong. maybe you have this process documented / drawn / diagrammed somewhere?

Reply:
We have a diagram from a presentation last quarter, but it needs an explanation to go along with it. I'll try to make a more in detail one and send it your way.

Reply:
So what we have decided to do is at each vote, the sender will build the vote without the signature, then hash and sign the vote(including previous vote chains) and attach that signature. That way we have the raw data to work with and traverse easily while still being able to verify the entire chain when need be by unsigning and verifying. We would need to implement the sorting since without ordering the hash could end up looking different for each person.